### PR TITLE
NM-96 Fix `RSCDomain` and `RSCRecord`'s `save()` methods to properly …

### PR DESCRIPTION
**Pull request by stoutput**
…expect an `OpenCloud\DNS\AsyncResponse` from the `Create()` call, and handle result appropriately

### CODE REVIEW

**Comment by stoutput** [comment]
Do not merge, pending QA

File: Model/RSCRecord.php [hunk 1]
```diff
@@ -166,47 +155,53 @@ public function save($data = null, $validate = true, $fieldList = array()) {
...
 		if (!$this->DNS) {
-			$this->_erorr('Unable to connect to DNS.');
+			$this->_error('Unable to connect to DNS.');
 		}
 		if (empty($name) || $zone === true || empty($zone)) {
 			$this->_error('zone and name are required to delete a record');
 		}
-		if (!$this->recordExists($name, $zone)) {
-			$this->_erorr("$name doesn't exist on $zone DNS.");
+		$Record = $this->__getRecordByNameZoneType($name, $zone, $type);
```
Comment (drewgg) [inline Model/RSCRecord.php:201]: Small nitpick about comments/annotations (that existed before your changes so not even about your code really):

Should the $zone type be changed from `string` to `mixed` or `string|true`, considering the default is boolean `true` and there is a `$zone === true` check in the method? You could move that "true is for strict compliance " comment to the $zone annotation as well:

`* @param string|true $zone Zone name; TRUE (default) is for strict compliance`

**Review by drewgg — APPROVED** [review]

**Review by webtechnick — APPROVED** [review]
Changes look good, thanks for finding and correcting that error misspellings. Kind of funny in an ironic way.

File: Model/RSCRecord.php (hunk 1 above)
Comment (stoutput) [reply Model/RSCRecord.php:201]: I would change this but honestly this whole package is full of docblock errors, that it's not really worth it to go through and fix it right now, and update all the references, re-merge, re-push, etc.

**Pull request merged**